tests/jballerina-integration-test/src/test/resources/configurables/configStructuredTypesProject/configStructuredTypes/modules/mod3/mod3.bal

```diff
@@ -0,0 +1,141 @@
+    "{\"arrVal\":[8.9,0.1],\"intVal\":33,\"stringVal\":\"ghi\",\"floatVal\":88.44,\"id\":103," + 
+    "\"mapVal\":{\"e\":\"e\",\"f\":789},\"mapArr\":[{\"g\":\"g\",\"h\":876}]}]");
+    test:assertEquals(memberTable.toString(), "[{\"id\":104,\"arrVal\":[\"4\",\"6\"],\"intVal\":65," + 
+    "\"stringVal\":\"val\",\"floatVal\":9.12,\"mapVal\":{\"c\":\"ccc\",\"d\":43}," + 
+    "\"mapArr\":[{\"i\":\"iii\",\"j\":21}]},{\"id\":105,\"arrVal\":[10,20],\"intVal\":40,\"stringVal\":\"str\"," + 
+    "\"floatVal\":13.57,\"mapVal\":{\"m\":\"mmm\",\"n\":68},\"mapArr\":[{\"y\":\"yyy\",\"x\":24}]}]");
+    test:assertEquals(memberMap.toString(), "{\"entry1\":{\"id\":106,\"arrVal\":[1,2,3],\"intVal\":22," + 
+    "\"stringVal\":\"abc\",\"floatVal\":22.33,\"mapVal\":{\"a\":\"a\",\"b\":123}},\"entry2\":{\"id\":107," + 
+    "\"mapArr\":[{\"c\":\"c\",\"d\":456}]}}");
+    test:assertEquals(memberMapTable.toString(), "{\"entry1\":[{\"id\":108,\"arrVal\":[100,200],\"intVal\":100," + 
+    "\"stringVal\":\"string\",\"floatVal\":103.507,\"mapVal\":{\"m\":\"value\",\"n\":608}," + 
+    "\"mapArr\":[{\"y\":\"test\",\"x\":204}]}]}");
+
+    testTableIterator(ownerTable);
+    testTableIterator(memberTable);
+
+    testMapIterator(productMapTable, 1);
+    testMapIterator(ownerMap, 2);
+    testMapIterator(ownerMapTable, 1);
+    testMapIterator(memberMap, 2);
+    testMapIterator(memberMapTable, 1);
+
+    // These lines should be enabled after fixing #30566
+    // testTableIterator(productTable);
+    // testMapIterator(productMap, 2);
+
+    testRestField();
+}
+
+public function testTableIterator(table<map<anydata>> tab) {
+    int count = 0;
+    foreach var entry in tab {
+        count += 1;
+    }
+    test:assertEquals(count, 2);
+}
+
+public function testMapIterator(map<anydata> testMap, int length) {
+    int count = 0;
+    foreach var entry in testMap {
+        count += 1;
+    }
+    test:assertEquals(count, length);
+}
+
+function testRestField() {
+    test:assertEquals(player.toString(), "{\"id\":101,\"teamMate\":{\"id\":102,\"teamMate\":{\"id\":103}}}");
+    test:assertEquals(playerArr.toString(), "[{\"teamMate\":{\"id\":105,\"teamMate\":{\"id\":106}},\"id\":104}," +
+    "{\"teamMate\":{\"id\":108,\"teamMate\":{\"id\":109}},\"id\":107}]");
+    test:assertEquals(playerTable.toString(), "[{\"id\":110,\"teamMate\":{\"id\":111,\"teamMate\":{\"id\":112}}}," +
+    "{\"id\":113,\"teamMate\":{\"id\":114,\"teamMate\":{\"id\":115}}}]");
+    test:assertEquals(playerMap.toString(), "{\"entry1\":{\"id\":116,\"teamMate\":{\"id\":117," +
+    "\"teamMate\":{\"id\":118}}},\"entry2\":{\"id\":119}}");
+    test:assertEquals(playerMapTable.toString(), "{\"entry1\":[{\"id\":120},{\"id\":121," +
+    "\"teamMate\":{\"id\":122,\"teamMate\":{\"id\":123}}}]}");
+
+    testTableIterator(playerTable);
+    testMapIterator(playerMap, 2);
+    testMapIterator(playerMapTable, 1);
+}
```

Comment:
Shall we create a issue for following unchanges files in PR?

Reply:
created https://github.com/ballerina-platform/ballerina-lang/issues/31490